File: 240410/흰검 칠하기/painting-white-black.py
N = int(input())

colored = [0] * 200001
cur_dir = 100000

white, black = [0] * 200001, [0] * 200001
w, b, g = 0, 0, 0


for _ in range(N):
    size, mv_dir = input().split()
    size = int(size)

    if mv_dir == 'R':
        for i in range(cur_dir, cur_dir + size):
            colored[i] = 2
            black[i] += 1
        cur_dir += size - 1
    else:
        for i in range(cur_dir - size + 1, cur_dir + 1):
            colored[i] = 1
            white[i] += 1
        cur_dir -= size - 1

for i in range(len(colored)):
    if white[i] >= 2 and black[i] >= 2:
        g += 1
    elif colored[i] == 1:
        w += 1
    elif colored[i] == 2:
        b += 1

print(w, b, g)

File: 260120/흰검 칠하기/painting-white-black.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N;
    cin >> N;

    const int SIZE = 200001;
    vector<int> colored(SIZE, 0);          
    vector<int> white(SIZE, 0), black(SIZE, 0);

    int cur_dir = 100000;
    long long w = 0, b = 0, g = 0;

    for (int t = 0; t < N; t++) 
    {
        int size;
        char mv_dir;
        cin >> size >> mv_dir;

        if (mv_dir == 'R') 
        {
            for (int i = cur_dir; i < cur_dir + size; i++) 
            {
                colored[i] = 2;
                black[i] += 1;
            }
            cur_dir += size - 1;
        } 
        else 
        { 
            for (int i = cur_dir - size + 1; i <= cur_dir; i++) 
            {
                colored[i] = 1;
                white[i] += 1;
            }
            cur_dir -= size - 1;
        }
    }

    for (int i = 0; i < SIZE; i++) 
    {
        if (white[i] >= 2 && black[i] >= 2) g++;
        else if (colored[i] == 1) w++;
        else if (colored[i] == 2) b++;
    }

    cout << w << " " << b << " " << g << "\n";
    return 0;
}